ui/src/app/components/object-detail/object-detail.component.html

```diff
@@ -112,208 +112,278 @@ <h3 class="title">
+              <td
+                mat-cell
+                *matCellDef="let element"
+                [ngClass]="{ 'drop-button-left-border': isEditMode }"
+              >
+                <div
+                  class="actions"
+                  *ngIf="
+                    isEditMode &&
+                    currentObject!.isSpannerNode &&
+                    !(element.get('spOrder').value === '')
+                  "
                 >
-                  <path
-                    fill-rule="evenodd"
-                    clip-rule="evenodd"
-                    d="M4 8C5.86384 8 7.42994 6.72523 7.87398 5H10V7H13V5H14V3H7.87398C7.42994 1.27477 5.86384 0 4 0C1.79086 0 0 1.79086 0 4C0 6.20914 1.79086 8 4 8ZM6 4C6 5.10457 5.10457 6 4 6C2.89543 6 2 5.10457 2 4C2 2.89543 2.89543 2 4 2C5.10457 2 6 2.89543 6 4Z"
-                    fill="black"
-                  />
-                </svg>
-              </div>
-            </td>
-          </ng-container>
-
-          <ng-container matColumnDef="spIsNotNull">
-            <th mat-header-cell class="table_header" *matHeaderCellDef>
-              <span *ngIf="isEditMode">Not Null</span><span *ngIf="!isEditMode">Nullable</span>
-            </th>
-            <td mat-cell *matCellDef="let element">
-              <div class="notNullContent">
-                <div *ngIf="isEditMode && element.get('spOrder').value !== ''">
-                  <mat-checkbox
-                    color="primary"
-                    [formControl]="element.get('spIsNotNull')"
-                  ></mat-checkbox>
+                  <mat-icon [matMenuTriggerFor]="menu">more_vert</mat-icon>
+                  <mat-menu #menu="matMenu" xPosition="before">
+                    <button mat-menu-item (click)="dropColumn(element)">
+                      <span>Drop Column</span>
+                    </button>
+                  </mat-menu>
                 </div>
-                <p *ngIf="!isEditMode">
-                  {{ element.get('spIsNotNull').value ? 'Not Null' : '' }}
-                </p>
-              </div>
-            </td>
-          </ng-container>
+              </td>
+            </ng-container>
 
-          <ng-container matColumnDef="dropButton">
-            <th mat-header-cell class="table_header" *matHeaderCellDef></th>
-            <td
-              mat-cell
-              *matCellDef="let element"
-              [ngClass]="{ 'drop-button-left-border': isEditMode }"
+            <tr mat-header-row *matHeaderRowDef="['spDatabase']"></tr>
+            <tr mat-header-row *matHeaderRowDef="spDisplayedColumns"></tr>
+            <tr mat-row *matRowDef="let row; columns: spDisplayedColumns"></tr>
+          </table>
+        </div>
+
+        <div [formGroup]="addColumnForm" *ngIf="isEditMode" class="add-column">
+          <mat-form-field appearance="outline" class="column-list">
+            <mat-label>Column Name</mat-label>
+            <mat-select
```

Comment:
Modifying the column name in a table is modifying the associated referred column of the other table too.
To reproduce - Modify the column name of a column associated to foreign key

Reply:
Basically editing a column name user_id is modifying the columns of all the tables having name as user_id

Reply:
Similar thing happens after interleaving the table and making column edit

Reply:
Updated. Please review it.